spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
Absolutely agreed. Just to clarify, I'm not against injecting `GoogleCredentials` and using it in the adapters.

My only concern is that it might be confusing for users that project ID is gotten from `GcpProperties`, which also contains a `credentials` named field, but they must get the credentials from somewhere else.

Another way to resolve this confusion is name the field `credentialsLocation` and document how channel adapters must be created.